expertsystem/data.py

```diff
@@ -375,7 +390,7 @@ def find(self, search_term: Union[int, str]) -> Particle:
         if isinstance(search_term, int):
             pid = search_term
             search_results = [
-                particle for particle in self.values() if particle.pid == pid
+                particle for particle in self if particle.pid == pid
```

Comment:
I'm thinking if this `find` can be combined with `filter`.

Reply:
I prefer not, because `find` returns a `Particle` instance whereas `filter` returns a `ParticleCollection`.

Reply:
I know, but I think that definition of `find` is confusing as it will raise an exception if more than one candidate is found. I would expect to get all results or the first occurrence. I prefer all results. The return type can also `Union[Particle, List[Particle]]` for convenience.

Reply:
I greatly to keep the return type unambiguous. But yes, the exception for duplicate PIDs is dubious. Actually that should already be checked when adding, because we want the PIDs to be unique (for instance because it's used as a check in the conservation rules).

Reply:
> Actually that should already be checked when adding, because we want the PIDs to be unique (for instance because it's used as a check in the conservation rules).

Right. It should actually be the other way around, that searching using a string can return multiple candidates.

Reply:
I'm still a bit hesitant to touch the `find` method. Currently, it is guaranteed that the names unique (if not, the instance is overwritten and a warning is printed). For PID there is now also such a warning, but there can still exist a situation where there are two equivalent particle with the same PID. That's why an exception is raised there.

Reply:
I think now with #295 this makes more sense as the str and pid are more sorting attributes of the `ParticleCollection` class now. The pid should also be handled in a unique way by the ParticleCollection.

Reply:
So PID and name are handled on an equal footing now, I think 3951fd64